back-end/src/reservations/reservations.controller.js

```diff
@@ -217,6 +219,30 @@ async function create(req, res) {
   res.status(201).json({ data: create[0] });
 }
 
+async function edit(req, res) {
+  const {
+    data: {
+      first_name,
+      last_name,
+      mobile_number,
+      reservation_date,
+      reservation_time,
+      people,
+    } = {},
+  } = req.body;
+  const updatedReservation = {
+    ...res.locals.reservation,
```

Comment:
Why are we spreading the object here? Dont we basically set the entire object again underneath? Is the spread necessary?